mailbox/api/src/main/java/org/apache/james/mailbox/events/MailboxListener.java

```diff
@@ -77,7 +82,17 @@ default boolean isHandling(Event event) {
      *
      * @param event not null
      */
-    void event(Event event) throws Exception;
+    default void event(Event event) throws Exception {
```

Comment:
this strategy has too big an impact in my opinion: we could have two distinct interfaces and wrap regular Listeners into ReactiveListeners when we register them in the bus. WDYT?

Reply:
Sounds like a lot of boiler plate for some behavior that could be overloaded as need.

Reply:
(I don't see the added value)

The down side is poluting the eventBus API, that is already dense.

Reply:
it's not, you will wrap at a single location vs what you did that is actually more boilerplate.

Reply:
We can bet comparing lines of code of both versions ;-)